nodes/IfNode.java

```diff
@@ -54,7 +55,22 @@ public String convertToPython() {
 
     @Override
     public boolean validateTree() {
-        return false;
+
+        if(expr.getType() != Types.BOOLEAN){
+            throw(new Exception()); //TODO Semantic Exception
+            return false;
+        }
+        if(!body.validateTree()){
+            return false;
+        }
+
+        for(ElseifNode e : elifLst) {
+            if (!e.validateTree()) {
+                return false;
```

Comment:
please add `e.setHasIfParent(true);` right below the `!e.validateTree()`  if condition! This will help me with the elseIfNode validation